testing/test_hookrelay.py

```diff
@@ -76,3 +76,32 @@ def hello(self, arg):
     pm.register(Plugin())
     res = pm.hook.hello(arg=3)
     assert res == 4
+
+
+def test_defaults(pm):
+    """Verify that default keyword arguments can be declared on both specs
+    and impls. The default value look up precedence is up as follows:
+        - caller provided value
+        - hookspec default
+        - hookimpl default
+    """
+    class Api:
+        @hookspec
+        def myhook(self, arg, kwarg="default"):
```

Comment:
Looking at this test I kind agree with the idea of only allowing `None` as defaults in specs and impls. This is would be less confusing to users and doesn't prevent us to actually allow other values in the future if one of them becomes "obvious".